### PR TITLE
Fgr 3641 shipping speed

### PR DESCRIPTION
**Pull request by Mikewa33**
Test incoming to cover shipping adjustment and selection

### CODE REVIEW

File: app/models/spree_mirakl/api.rb [hunk 1]
```diff
@@ -19,13 +19,17 @@ def account
   end
 
   def carriers
-    @response = SpreeMirakl::Request.new(@store).get("/api/shipping/carriers?shop_id=#{@store.shop_id}")
+    @response = SpreeMirakl::Request.new(@store).get("/api/shipping/carriers")
```
[the remark this answers is not in the document]
Comment (Mikewa33) [reply app/models/spree_mirakl/api.rb:22]: This was removed cause when it is called we won't have the shop id yet

[the remark this answers is not in the document]
Comment (Mikewa33) [reply app/models/spree_mirakl/api.rb:22]: It isn't needed for the call

File: app/services/mirakl/build_order.rb [hunk 2]
```diff
@@ -72,6 +88,19 @@ def line_items_hash(order_lines)
     def build_order_for_user(order_data, store)
       @order = Spree::Core::Importer::Mirakl::Order.import(store.user, get_order_hash(order_data, store))
       @order.shipments.each(&:finalize!) # This is required to decrease inventory
+      order_shipping_adjustment(@order, store, order_data)
```
[the remark this answers is not in the document]
Comment (Mikewa33) [reply app/services/mirakl/build_order.rb:97]: We would go with the default and let it build a shipment normally and then just change the shipping method afterward?

[the remark this answers is not in the document]
Comment (hf2186) [reply app/services/mirakl/build_order.rb:97]: Yeah -> Update Shipping Method -> Update payment and instead remove adjustment (Adj may cause issues on refund but not sure)

[the remark this answers is not in the document]
Comment (Mikewa33) [reply app/services/mirakl/build_order.rb:97]: Shipping refund price comes from mirakl when a refund is processed so it shouldn't affect that. We do need to determine if we are refunding shipping on returns that pay extra for shipping(we may have to depending on the store)

[the remark this answers is not in the document]
Comment (Mikewa33) [reply app/services/mirakl/build_order.rb:97]: After trying it out it is easier to have it go through the order importer. If we do it after then we have to deal with reselecting the shipping method and recalculating the rates

**Pull request merged**